gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterMetricsStatsExporter.java

```diff
@@ -0,0 +1,142 @@
+{
+    private static final Logger log = Logger.get(ClusterMetricsStatsExporter.class);
+
+    private final MBeanExporter exporter;
+    private final GatewayBackendManager gatewayBackendManager;
+    private final Duration refreshInterval;
+    // MBeanExporter uses weak references, so clustersStats Map is needed to maintain strong references to metric objects to prevent garbage collection
+    private final Map<String, ClusterMetricsStats> clustersStats = new HashMap<>();
+    private final ScheduledExecutorService scheduledExecutor = Executors.newSingleThreadScheduledExecutor();
+
+    @Inject
+    public ClusterMetricsStatsExporter(GatewayBackendManager gatewayBackendManager, MBeanExporter exporter, MonitorConfiguration monitorConfiguration)
+    {
+        this.gatewayBackendManager = requireNonNull(gatewayBackendManager, "gatewayBackendManager is null");
+        this.exporter = requireNonNull(exporter, "exporter is null");
+        this.refreshInterval = monitorConfiguration.getClusterMetricsRegistryRefreshPeriod();
+    }
+
+    @PostConstruct
+    public void start()
+    {
+        log.debug("Running periodic metric refresh with interval of %s", refreshInterval);
+        scheduledExecutor.scheduleAtFixedRate(() -> {
+            try {
+                updateClustersMetricRegistry();
+            }
+            catch (Exception e) {
+                log.error(e, "Error refreshing cluster metrics");
+            }
+        }, 0, refreshInterval.toMillis(), MILLISECONDS);
+    }
+
+    @PreDestroy
+    public void stop()
+    {
+        scheduledExecutor.shutdownNow();
+    }
+
+    private synchronized void updateClustersMetricRegistry()
+    {
+        // Get current clusters from DB
+        Set<String> currentClusters = gatewayBackendManager.getAllBackends().stream()
+                .map(ProxyBackendConfiguration::getName)
+                .collect(Collectors.toSet());
+
+        // Create a copy of keys to avoid concurrent modification
+        Set<String> registeredClusters = new HashSet<>(clustersStats.keySet());
+
+        // Unregister metrics for removed clusters
+        for (String registeredCluster : registeredClusters) {
+            if (!currentClusters.contains(registeredCluster)) {
+                try {
+                    exporter.unexportWithGeneratedName(ClusterMetricsStats.class, registeredCluster);
+                    log.debug("Unregistered metrics for removed cluster: %s", registeredCluster);
+                    clustersStats.remove(registeredCluster);
+                }
+                catch (Exception e) {
+                    log.error(e, "Failed to unregister metrics for cluster: %s", registeredCluster);
+                }
+            }
+        }
+
+        // Register metrics for added clusters
+        for (String cluster : currentClusters) {
+            if (!clustersStats.containsKey(cluster)) {
+                registerClusterMetrics(cluster);
+            }
+        }
+    }
+
```

Comment:
How about exposing a package-private method to trigger updateClustersMetricRegistry directly for tests. 
```suggestion
// Visible for testing
synchronized void updateClustersMetricRegistry()
```